service/src/main/kotlin/app/cash/backfila/service/runner/statemachine/BatchAwaiter.kt

```diff
@@ -42,8 +44,7 @@ class BatchAwaiter(
 
       // Repeat this batch until it succeeds.
       retry@ while (true) {
-        var retry: Deferred<RunBatchResponse>? = null
-        try {
+        val nextRequest: Deferred<RunBatchResponse> = try {
```

Comment:
I love kotlin

Reply:
oh I think this is worse, just because there’s so much code between the declaration on line 46 and the value on line 100

Reply:
might read better to extract this whole thing into a method?

Reply:
The good thing is that the robots are making sure I don't miss to return a value. Easy to understand... fair enough.

Reply:
A method won't work because of the way it breaks out of the various loops. We can talk about it tomorrow during our pairing session perhaps?